---
layout: page
menu: true
permalink: /publications/
title: Publications
---


- **Accounting for errors in data improves timing in single-cell cancer evolution**<br>
   Kylie Chen, Jiří Moravec, Alex Gavryushkin, David Welch, and Alexei Drummond<br>
  *bioRxiv* 2021<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.03.17.435906)

- **A Fast Lasso-based Method for Inferring Pairwise Interactions**<br>
  Kieran Elmes, Astra Heywood, Zhiyi Huang, and Alex Gavryushkin<br>
  *bioRxiv* 2021<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.12.13.471844)

- **Cancer phylogenetics using single-cell RNA-seq data**<br>
  Jiří Moravec, Rob Lanfear, David Spector, Sarah Diermeier, and Alex Gavryushkin<br>
  *bioRxiv* 2021<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.01.07.425804)

- **Tuberous sclerosis complex: a complex case**<br>
  Ryan Powell, Sharon Pattison, Jiri Moravec, Basharat Bhat, Nada Guirguis, David Markie, Greg Jones, Jason Copedo, Ian Morison, Alex Gavryushkin, Bronwyn Gray, Lisa Wyeth, Mike Eccles, and Erin Macaulay<br>
  *Cold Spring Harbor Molecular Case Studies,* 8: a006182, 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1101/mcs.a006182)

- **Sustained software development, not number of citations or journal choice, is indicative of accurate bioinformatic software**<br>
   Paul Gardner, James Paterson, Stephanie McGimpsey, Fatemeh Ashari-Ghomi, Sinan Umu, Aleksandra Pawlik, Alex Gavryushkin, and Michael Black<br>
  *Genome Biology,* 23(56), 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1186/s13059-022-02625-x)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/092205)

- **Discrete Coalescent Trees**<br>
  Lena Collienne, Kieran Elmes, Mareike Fischer, David Bryant, and Alex Gavryushkin<br>
  *Journal of Mathematical Biology,* 83(60), 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-021-01685-0)
  [<i class="ai ai-arxiv-square ai-3x"></i>](https://arxiv.org/abs/2101.02751)

- **Learning epistatic gene interactions from perturbation screens**<br>
  Kieran Elmes, Fabian Schmich, Ewa Szczurek, Jeremy Jenkins, Niko Beerenwinkel, and Alex Gavryushkin<br>
  *PLoS ONE,* 16(7), e0254491, 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1371/journal.pone.0254491)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2020.08.24.264713)

- **Computing nearest neighbour interchange distances between ranked phylogenetic trees**<br>
  Lena Collienne and Alex Gavryushkin<br>
  *Journal of Mathematical Biology,* 82(8), 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-021-01567-5)
  [<i class="ai ai-arxiv-square ai-3x"></i>](https://arxiv.org/abs/2007.12307)

- **Geometry of Ranked Nearest Neighbour Interchange Space of Phylogenetic Trees**<br>
  Lena Collienne, Kieran Elmes, Mareike Fischer, David Bryant, and Alex Gavryushkin<br>
  *bioRxiv* 2019<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2019.12.19.883603)
  [<i class="ai ai-3x">blog</i>]({% post_url 2019-12-23-lenas-rnni-convextiy-paper %})

- **High-dimensional microbiome interactions shape host fitness**<br>
  Alison Gould, Vivian Zhang, Lisa Lamberti, Eric Jones, Benjamin Obadia, Nikolaos Korasidis, Alex Gavryushkin, Jean Carlson, Niko Beerenwinkel, and Will Ludington<br>
  *PNAS,* 115(51), E11951-E11960, 2018<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1073/pnas.1809349115)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/232959)

- **The geometry of partial fitness orders and an efficient method for detecting genetic interactions**<br>
  Caitlin Lienkaemper, Lisa Lamberti, James Drain, Niko Beerenwinkel, and Alex Gavryushkin<br>
  *Journal of Mathematical Biology,* 77(4), 951–970, 2018<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-018-1237-7)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/180976)

- **Inferring genetic interactions from comparative fitness data**<br>
  Kristina Crona, Alex Gavryushkin, Devin Greene, and Niko Beerenwinkel<br>
  *eLife,* 6:e28629, 2017<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.7554/eLife.28629)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/137372)

- **The combinatorics of discrete time-trees: theory and open problems**<br>
  Alex Gavryushkin, Chris Whidden, and Erick Matsen<br>
  *Journal of Mathematical Biology,* 76(5), 1101–1121, 2017<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-017-1167-9)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/063362)
  [<i class="ai ai-3x">blog</i>](http://matsen.fredhutch.org/general/2016/07/11/discrete-time-tree.html) post by Erick

- **The space of ultrametric phylogenetic trees**<br>
  Alex Gavryushkin and Alexei Drummond<br>
  *Journal of Theoretical Biology,* 403, 197–208, 2016<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1016/j.jtbi.2016.05.001)
  [<i class="ai ai-arxiv-square ai-3x"></i>](http://arxiv.org/abs/1410.3544)
