---
layout: page
menu: true
permalink: /publications/
title: Publications
---


- **Filtering for highly variable genes and high quality spots improves phylogenetic analysis of cancer spatial transcriptomics Visium data**<br>
   Sasha Gavryushkina, Holly Pinkney, Sarah Diermeier, Alex Gavryushkin<br>
  *bioRxiv* 2024<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2024.07.11.603166)

- **The Fossilised Birth-Death Model is Identifiable**<br>
  Kate Truman, Timothy Vaughan, Alex Gavryushkin, and Sasha Gavryushkina<br>
  *Systematic Biology* 2024 (to appear)<br>
  *bioRxiv* 2024<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2024.02.08.579547)

- **A tractable tree distribution parameterized by clade probabilities and its application to Bayesian phylogenetic point estimation**<br>
   Lars Berling, Jonathan Klawitter, Remco Bouckaert, Dong Xie, Alex Gavryushkin, and Alexei Drummond<br>
  *bioRxiv* 2024<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2024.02.20.581316)

- **Automated convergence diagnostic for phylogenetic MCMC analyses**<br>
   Lars Berling, Remco Bouckaert, and Alex Gavryushkin<br>
   *IEEE/ACM Transactions on Computational Biology and Bioinformatics* 2024 (to appear)<br>
  *bioRxiv* 2023<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2023.08.10.552869)

- **Online Bayesian analysis with BEAST 2**<br>
   Remco Bouckaert, Lena Collienne, and Alex Gavryushkin<br>
  *bioRxiv* 2022<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2022.05.03.490538)

- **Estimating the mean in the space of ranked phylogenetic trees**<br>
   Lars Berling, Lena Collienne, and Alex Gavryushkin<br>
  *Bioinformatics* 2024<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1093/bioinformatics/btae514)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2023.05.08.539790)

- **Ranked subtree prune and regraft**<br>
  Lena Collienne, Chris Whidden, and Alex Gavryushkin<br>
  *Bulletin of Mathematical Biology* 2024<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s11538-023-01244-2)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2023.05.16.541039)

- **Testing for phylogenetic signal in single-cell RNA-seq data**<br>
  Jiří Moravec, Rob Lanfear, David Spector, Sarah Diermeier, and Alex Gavryushkin<br>
  *Journal of Computational Biology* 2023<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](http://doi.org/10.1089/cmb.2022.0357)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.01.07.425804)

- **A fast lasso-based method for inferring higher-order interactions**<br>
  Kieran Elmes, Astra Heywood, Zhiyi Huang, and Alex Gavryushkin<br>
  *PLOS Computational Biology* 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1371/journal.pcbi.1010730)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.12.13.471844)

- **SNVformer: an attention-based deep neural network for GWAS data**<br>
   Kieran Elmes, Diana Benavides-Prado, Neset Ozkan Tan, Trung Bao Nguyen, Nicholas Sumpter, Megan Leask, Michael Witbrock, and Alex Gavryushkin<br>
  *ICML Workshop on Computational Biology* 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://icml-compbio.github.io/2022/papers/WCBICML2022_paper_58.pdf)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2022.07.07.499217)

- **Accounting for errors in data improves timing in single-cell cancer evolution**<br>
   Kylie Chen, Jiří Moravec, Alex Gavryushkin, David Welch, and Alexei Drummond<br>
  *Molecular Biology and Evolution* 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1093/molbev/msac143)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2021.03.17.435906)

- **Tuberous sclerosis complex: a complex case**<br>
  Ryan Powell, Sharon Pattison, Jiri Moravec, Basharat Bhat, Nada Guirguis, David Markie, Greg Jones, Jason Copedo, Ian Morison, Alex Gavryushkin, Bronwyn Gray, Lisa Wyeth, Mike Eccles, and Erin Macaulay<br>
  *Cold Spring Harbor Molecular Case Studies* 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://www.ncbi.nlm.nih.gov/pmc/articles/pmc9059781/)

- **Sustained software development, not number of citations or journal choice, is indicative of accurate bioinformatic software**<br>
   Paul Gardner, James Paterson, Stephanie McGimpsey, Fatemeh Ashari-Ghomi, Sinan Umu, Aleksandra Pawlik, Alex Gavryushkin, and Michael Black<br>
  *Genome Biology* 2022<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1186/s13059-022-02625-x)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/092205)

- **Discrete coalescent trees**<br>
  Lena Collienne, Kieran Elmes, Mareike Fischer, David Bryant, and Alex Gavryushkin<br>
  *Journal of Mathematical Biology* 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-021-01685-0)
  [<i class="ai ai-arxiv-square ai-3x"></i>](https://arxiv.org/abs/2101.02751)

- **Learning epistatic gene interactions from perturbation screens**<br>
  Kieran Elmes, Fabian Schmich, Ewa Szczurek, Jeremy Jenkins, Niko Beerenwinkel, and Alex Gavryushkin<br>
  *PLOS ONE* 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1371/journal.pone.0254491)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2020.08.24.264713)

- **Computing nearest neighbour interchange distances between ranked phylogenetic trees**<br>
  Lena Collienne and Alex Gavryushkin<br>
  *Journal of Mathematical Biology* 2021<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-021-01567-5)
  [<i class="ai ai-arxiv-square ai-3x"></i>](https://arxiv.org/abs/2007.12307)

- **Geometry of ranked nearest neighbour interchange space of phylogenetic trees**<br>
  Lena Collienne, Kieran Elmes, Mareike Fischer, David Bryant, and Alex Gavryushkin<br>
  *bioRxiv* 2019<br>
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/2019.12.19.883603)
  [<i class="ai ai-3x">blog</i>]({% post_url 2019-12-23-lenas-rnni-convextiy-paper %})

- **High-dimensional microbiome interactions shape host fitness**<br>
  Alison Gould, Vivian Zhang, Lisa Lamberti, Eric Jones, Benjamin Obadia, Nikolaos Korasidis, Alex Gavryushkin, Jean Carlson, Niko Beerenwinkel, and Will Ludington<br>
  *PNAS* 2018<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1073/pnas.1809349115)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/232959)

- **The geometry of partial fitness orders and an efficient method for detecting genetic interactions**<br>
  Caitlin Lienkaemper, Lisa Lamberti, James Drain, Niko Beerenwinkel, and Alex Gavryushkin<br>
  *Journal of Mathematical Biology* 2018<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-018-1237-7)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/180976)

- **Inferring genetic interactions from comparative fitness data**<br>
  Kristina Crona, Alex Gavryushkin, Devin Greene, and Niko Beerenwinkel<br>
  *eLife* 2017<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.7554/eLife.28629)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/137372)

- **The combinatorics of discrete time-trees: theory and open problems**<br>
  Alex Gavryushkin, Chris Whidden, and Erick Matsen<br>
  *Journal of Mathematical Biology* 2017<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1007/s00285-017-1167-9)
  [<i class="ai ai-biorxiv-square ai-3x"></i>](https://doi.org/10.1101/063362)
  [<i class="ai ai-3x">blog</i>](https://matsen.fredhutch.org/general/2016/07/11/discrete-time-tree.html) post by Erick

- **The space of ultrametric phylogenetic trees**<br>
  Alex Gavryushkin and Alexei Drummond<br>
  *Journal of Theoretical Biology* 2016<br>
  [<i class="ai ai-open-access-square ai-3x"></i>](https://doi.org/10.1016/j.jtbi.2016.05.001)
  [<i class="ai ai-arxiv-square ai-3x"></i>](https://arxiv.org/abs/1410.3544)
