<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">
	<title>Somatic Phylogenetics - Alex Gavryushkin</title>
	<meta name="author" content="Alex Gavryushkin">
	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
	<link rel="stylesheet" href="css/reveal.css">
	<link rel="stylesheet" href="css/theme/simple_gavruskin.css" id="theme">
	<link rel="stylesheet" href="lib/css/zenburn.css">
	<link rel="icon" href="../favicon.ico" />
	<!--[if lt IE 9]>
	<script src="lib/js/html5shiv.js"></script>
	<![endif]-->
</head>

<body>
<div class="reveal">
<div class="slides">

<section>
	<h3>
	  Somatic Phylogenetics
	</h3>
	<br>
	<h4>
	  Alex Gavryushkin<br>
	  <img data-src="images/otago_university_logo.jpg" width=25%>
	</h4>
	<br>
	<p style="font-size:14pt">
	  6 July 2020
	</p>
</section>

<section>
        <h4>What this talk is NOT about</h4>
	<a href="https://biods.org">
          <img width="60%" data-src="images/funding_RDF_CS_bioDS.svg">
	</a>
	<br>
	<ul>
	  <li> <a href="https://biods.org/opportunities/">Phylogenomic methods and tools for within-patient tumour evolution analysis</a>
	  <li> <a href="https://biods.org/opportunities/">Evolutionary landscapes of microbial resistance</a>
	  <li> <a href="https://biods.org/opportunities/">Online algorithms in computational biology</a>
	  <li> We are hiring!
	</ul>
</section>

<section>
	<h3>Phylogenetic inference</h3>
	<img data-src="images/hyrax.svg" width="100%">
	<br>
	<div align="center">
	<img class="fragment" data-src="images/hyrax_foto.jpg" width="40%" data-fragment-index="1">
	</div>
	<p class="fragment" data-fragment-index="1" style="font-size:10pt" align="center">
	Image source: Wikipedia
	</p>
</section>

<section>
	<h3>
	  Why is it suddenly a thing?
	</h3>
	<img class="stretch" data-src="../teaching/images/costpergenome_2017.jpg">
</section>

<section>
	<h4>
	  Molecular biology: crash course
	</h4>
	<img class="stretch" data-src="../teaching/images/central_dogma.svg">
</section>

<section data-background-video="http://content.dnalc.org/content/c16/16933/3d-central-dogma-audio-title.mp4">
</section>

<section>
	<h3>Tumour evolution</h3>
	<img class="stretch" data-src="images/Katharina_trees.svg">
	<p style="font-size:10pt" align="center">
	Jahn, Katharina, Jack Kuipers, and Niko Beerenwinkel. "Tree inference for single-cell data." <i>Genome biology</i> 17.1 (2016): 86.
	</p>
</section>

<section>
	<h3>Timing tumour evolution</h3>
	<img class="stretch" data-src="images/Sottoriva_tumour_time_tree.png">
	<p style="font-size:10pt" align="center">
	Lote, H., I. Spiteri, L. Ermini, A. Vatsiou, A. Roy, A. McDonald, N. Maka, et al. 2017. "Carbon Dating Cancer: Defining the Chronology of Metastatic Progression in Colorectal Cancer." <i>Annals of Oncology</i> 28 (6): 1243–49.
	</p>
</section>

<section>
	<div style="font-size:10pt">
	<img data-src="images/Alexei_with_Tanja.svg" width="60%">
	<ul>
	  <li> Gavryushkin, Alex, and Alexei J. Drummond. "The space of ultrametric phylogenetic trees."
	  <i>Journal of Theoretical Biology</i> 403 (2016): 197-208.
	  <li> Stadler, Tanja, Timothy G. Vaughan, Alex Gavryushkin, Stephane Guindon, Denise Kühnert, Gabriel E. Leventhal, Alexei J. Drummond.
	  "How well can the exponential-growth coalescent approximate constant-rate birth-death population dynamics?"
	  <i>Proc. R. Soc. B.</i> Vol. 282. No. 1806. The Royal Society, 2015.
	<ul>
	<img data-src="images/t_tau_spaces.svg" width="75%">
	</div>
</section>

<section>
	<h4>tau-space</h4>
	<img class="stretch" data-src="images/tau_space.svg">
</section>

<section data-transition="slide-in fade-out">
	<h4>Geodesic in time-tree space</h4>
	<div style="font-size:10pt">
	<ul>
	  <li> Gavryushkin, Alex, and Alexei J. Drummond. "The space of ultrametric phylogenetic trees."
	  <i>Journal of Theoretical Biology</i> 403 (2016): 197-208.
	  <li> Gavryushkin, Alex, and Alexei Drummond. <i>tauGeodesic</i>. Mar. 2015. doi: 10.5281/zenodo.47152.
	<a href="https://github.com/gavruskin/tauGeodesic">https://github.com/gavruskin/tauGeodesic</a>
	</ul>
	<img data-src="images/trees.gif" width="60%">
	<div>
</section>

<section data-transition="fade-in slide-out" data-background-image="images/penguin_emperor_background.svg">
	<div style="font-size:10pt">
	<ul>
	  <li> Gavryushkina, Alexandra, Tracy A. Heath, Daniel T. Ksepka, Tanja Stadler,
	  David Welch, and Alexei J. Drummond.
	  "Bayesian total-evidence dating reveals the recent crown radiation of penguins."
	  <i>Systematic Biology</i> 66.1 (2016): 57-73.
	  <li> Gavryushkin, Alex, and Alexei Drummond. <i>tauGeodesic</i>. Mar. 2015. doi: 10.5281/zenodo.47152.
	<a href="https://github.com/gavruskin/tauGeodesic">https://github.com/gavruskin/tauGeodesic</a>
	</ul>
	<img data-src="images/trees_penguins.gif" width="60%">
	<div>
</section>

<section>
	<h4>t-space: simplex</h4>
	<img class="stretch" data-src="images/t_space_simplex.svg">
</section>

<section>
	<h4>t-space: 2D</h4>
	<img class="stretch" data-src="images/t_space_2D.svg">
</section>

<section>
	<h4>t-space: 3D</h4>
	<img class="stretch" data-src="images/t_space_3D.svg">
</section>

<section>
  <section>
	<h3>(Discrete) Timer-tree</h3>
	<img class="stretch" data-src="images/discreteTimeTree.pdf.png">
  </section>
  <section>
	<h3>Sampled ancestor tree</h3>
	<img class="stretch" data-src="images/SampledAncestorTree.svg">
  </section>
</section>

<section>
	<h3>NNI graph</h3>
	<img class="stretch" data-src="images/NNI.pdf.png">
</section>

<section>
  <section>
	<h3>Why is this important?</h3>
	<ul>
		<li> Tree search algorithms
		<li> Model testing/selection and other simulation studies
	</ul>
	<h3>
	<p class="fragment fade-in" data-fragment-index="2">
	Why is this hard?
	</p>
	</h3>
	<p class="fragment fade-in" data-fragment-index="3">
	Trees are many!
	<br>
	<img data-src="images/NNI_diameter.svg">
	</p>
  </section>
  <section data-markdown>
	<script type="text/template">
		## Let&#39;s look at the numbers
		| $\hskip10mm n \hskip10mm$ | $\hskip2cm n! \hskip2cm$ | $\hskip1cm n\log n \hskip1cm$ |
	        |:-------:|:--------------------:|:----------------------:|
        	|    10   |   3,628,800          |          34            |
	        |    20   |$2.4\times 10^{18}$   |          87            |
        	|    30   |$2.6\times 10^{32}$   |         148            |
		![](images/NNI_diameter.svg)
	</script>
  </section>
</section>

<section>
  <section>
	<h3>Discrete time-tree space</h3>
	<img class="stretch" data-src="images/DtT_with_rank_move.pdf.png">
  </section>
  <section>
	<h3>Sampled ancestor tree space</h3>
	<img class="stretch" data-src="images/SAT.svg">
  </section>
</section>

<section>
	<section>
	<h3>Trees at distance 2</h3>
	<img class="stretch" data-src="images/dts_neighbors_2.svg">
	</section>
	<section>
	<h3>Trees at distance 4</h3>
	<img class="stretch" data-src="images/dts_neighbors.svg">
	</section>
</section>

<section>
	<h3>Main idea</h3>
	<p>(my failed proof)</p>
	<img class="stretch" data-src="images/NNI_VS_rNNI.svg">
</section>

<section>
	<h3>History of the NNI graph</h3>
	<ul>
		<li> Over 25 year of work!
		<li> Over 7 erroneous papers published!
	</ul>
</section>

<section>
  <section>
	<h3>What's wrong with the NNI graph?</h3>
	<ol>
		<li>
		<p class="fragment fade-in" data-fragment-index="1">
		The Split Theorem
		</p>
		<li>
		<p class="fragment fade-in" data-fragment-index="2">
		The merge and sort trick
		</p>
	</ol>
  </section>
  <section data-transition="slide-in none-out">
	<h3>Merge and sort trick</h3>
	<img class="stretch" data-src="images/merge_and_sort_trick.svg">
  </section>
  <section data-transition="none-in slide-out">
	<h3>Merge and sort trick</h3>
	<img class="stretch" data-src="images/merge_and_sort_trick_solution.svg">
  </section>
</section>

<section>
  <section>
	<h3>RNNI is free from all these!</h3>
	<ul>
		<li> Split theorem. Tick.
		<li> Merge and sort doesn't work. Tick
		<li>
		<p class="fragment fade-in" data-fragment-index="1">
		Efficient polynomial algorithm?
		</p>
	</ul>
  </section>
  <section>
	<img class="stretch" data-src="images/we_want_you.svg">
  </section>
  <section>
	<h4>What is an approximate (?) algorithm?</h4>
	<img class="stretch" data-src="images/diameterUpperBound.svg">
  </section>
  <section>
	<h3>Graph grammars</h3>
	A graph grammar consists of a finite set of productions $\{L_i \to_i R_i\}$, where $L_i$ and $R_i$ are connected undirected edge-end labeled graphs and $\to_i$ is a one-to-one map between half-edges of $L_i$ and those of $R_i$.<br><br>
	The productions are then applied to the starting tree $T$ to derive all possible trees at $\mathrm{RNNI}$ distance up to $r$ from $T$.<br><br> 
	A production is said to be <i>ready</i> at a stage $s$ of the derivation if the running tree at stage $s$ has a subgraph isomorphic to the left side $L_i$ of the production.
  </section>
  <section>
	<h4>Graph grammar: example</h4>
	<img class="stretch" data-src="images/grammar_rNNIu.svg">
  </section>
  <section>
	<b>Theorem</b> (Sleator, Tarjan, and Thurston). Let $\mathcal{G}$ be a graph grammar and $G$ a graph on $n$ vertices.
	Then the number of trees at distance at most $r$ from $G$ is $(c+1)^{n+rm}$, where $c$ is the number of vertices in left sides of $\mathcal{G}$ and $m$ the maximum number of vertices in any right side of $\mathcal{G}$.<br>
	<img class="stretch" data-src="images/grammar_rNNIu.svg">
  </section>
  <section>
	<b>Theorem</b> (Sleator, Tarjan, and Thurston). Let $\mathcal{G}$ be a graph grammar and $G$ a graph on $n$ vertices.
	Then the number of trees at distance $r$ from $G$ is $(c+1)^{n+rm}$.<br><br>
	
	<b>Theorem</b> (Gavryushkin, Whidden, and Matsen). The number of trees within distance $r$ from any given tree is at most
	<br>
	$3^{n+2r−1}$ in $\mathrm{RNNIu}$&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
	$3^{2n+2r−1}$ in $\mathrm{RNNI}$<br>
	$4^{n+2r−1}$ in $\mathrm{DtT}$&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
	$4^{2n+2r−1}$ in $\mathrm{DtTu}$<br>
	<p class ="fragment fade-in" data-fragmen-index="1">
		<b>Corollary</b> (Gavryushkin, Whidden, and Matsen).
		$$\frac{1}{2} \log_3 \frac{(n-1)!n!}{6^{n-1}} \leq \mathrm{Diam}(\mathrm{RNNIu}) \leq n^2 - 3n - \frac{5}{8}$$
	</p>
  </section>
  <section>
	<h4>Proof:</h4>
	<img class="stretch" data-src="images/grammar_rNNIu.svg">
  </section>
</section>

<section>
	<h3>What we've done?</h3>
	<ul>
		<li>
		<p class="fragment fade-in" data-fragment-index="1">
		Introduced the RNNI graph on ranked trees (to the best of our knowledge)
		</p>
		<li>
		<p class="fragment fade-in" data-fragment-index="2">
		Established basic geometric properties of the graph
		</p>
		<li>
		<p class="fragment fade-in" data-fragment-index="3">
		Designed an efficient approximate algorithm for computing shortest paths
		</p>
		<li>
		<p class="fragment fade-in" data-fragment-index="4">
		Proved that all the fancy NNI methods, e.g. Sleator-Tarjan-Thurston and merge-and-sort argument, don't work
		</p>
		<li>
		<p class="fragment fade-in" data-fragment-index="5">
		<font color="blue">Failed to prove that RNNI is NP-hard</font>
		</p>
	</ul>
</section>

<section>
	<h3>What has to be done?</h3>
	<ul>
		<li> Is RNNI polynomial? Complexity?
		<li> Split Theorem
		<li> Are these two related?
		<li> ...
	</ul>
</section>


</div>
</div>

<script src="lib/js/head.min.js"></script>
<script src="js/reveal.js"></script>
<script>
  Reveal.initialize({
    dependencies: [
	// Speaker notes
	{ src: 'plugin/notes/notes.js', async: true },
        // Interpret Markdown in <section> elements
        { src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
        { src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
        // MathJax
        { src: 'plugin/math/math.js', async: true }
    ]
  });
  Reveal.configure({ slideNumber: true });
</script>
</body>
</html>
