<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">
	<title>
	  Algorithms for biomolecular data - Alex Gavryushkin
	</title>
	<meta name="author" content="Alex Gavryushkin">
	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
	<link rel="stylesheet" href="revealjs/dist/reveal.css">
	<link rel="stylesheet" href="revealjs_themes_biods/simple_gavruskin.css" id="theme">
	<link rel="stylesheet" href="revealjs/plugin/highlight/zenburn.css">
	<link rel="icon" href="../favicon.ico" />
</head>

<body>
<div class="reveal">
<div class="slides">

<section>
	<h3>
	  Scalable algorithms<br>
	  for biomolecular data
	</h3>
	<h4>
	  <a href="https://biods.org/alex/">Alex Gavryushkin</a>
	</h4>
	<h5>
	  <a href="https://rnafold.ai">
	    <img data-src="/assets/2025-02-19-FoldAI_logo.svg" width=60%>
	  </a>
	</h5>
	<font size="5">
	  Crash Course on Modern Biology at Macquarie University<br>
	  February 20, 2025
	</font>
</section>

<section>
	<h4>CS in Biology</h4>
	<img data-src="/assets/hassabis_jumper_nobel.svg" width=80%>
</section>

<section data-background-image="/assets/2025-02-18-isomorphic_labs_landing.png"
	data-background-size=80%>
</section>

<section>
	<h3>Three main* challenges for CS in biology</h3>
	<ol>
	  <li class="fragment" data-fragment-index="1">No ground truth</li>
	  <li class="fragment" data-fragment-index="2">No good loss functions</li>
	  <li class="fragment" data-fragment-index="3">Noise</li>
	</ol>
	<p class="fragment" data-fragment-index="4">
	  All versions of AlphaFold (so far) managed to avoid tackling any of these.
	</p>
	<font size="4">
	   *personal take
	</font>
</section>

<section data-background-image="/assets/2025-02-19-fold_ai_cover_slide.svg"
	data-background-size=60%>
</section>

<section data-background-image="/assets/2025-02-19-fold_ai_the_problem.svg"
	data-background-size=60%>
</section>

<section data-background-image="/assets/2025-02-19-fold_ai_the_solution.svg"
	data-background-size=60%>
</section>

<section data-background-image="/assets/2025-02-19-fold_ai_the_team.svg"
	data-background-size=60%>
</section>

<section>
	<h4>Supported by NSW Biosciences Fund</h4>
	<img data-src="/assets/biosf_ceremony.jpeg" width=80%>
</section>

<section>
	<h4>Model-based approaches to genotype-phenotype data</h4>
	<img data-src="/assets/fitland_cartoon.svg" width=50%>
	$
	  f(x_1,\, \ldots,\, x_p) = \beta_0 + \sum_{i} x_i \beta_i
	  + \sum_{i \le j} x_i x_j \beta_{i,j}
	$
	$
	  + \sum_{i \le j \le k} x_i x_j x_k \beta_{i,j,k}
	  + \ldots
	$<br>
</section>

<section>
	<h3>The linear model</h3>
	$
	  f(x_1,\, \ldots,\, x_p) = \beta_0 + \sum_{i} x_i \beta_i
	$
	<img class="fragment" data-src="/assets/95_ai_linear.png">
</section>

<section>
	<h4>Genome-Wide Association Studies</h4>
	<img data-src="https://upload.wikimedia.org/wikipedia/commons/4/4f/Manhattan_plot_from_a_GWAS_of_kidney_stone_disease.png"><br>
	<font size="3">
	  <a href="https://doi.org/10.1038/s41467-019-13145-x">
	  Sarah Howles et al. Genetic variants of calcium and vitamin D metabolism in kidney stone disease. 2019</a>
	</font>
</section>

<section>
	<img data-src="images/fitness_shapes_2_loci_BPS_axis.svg">
	<p align="left">
	  The set of genotypes is $\mathcal G = \{0,1\}^{p}$.
	</p>
	<p align="left">
	  A <i>fitness landscape</i> is a function $f:\mathcal G \to \mathbb R^+$.
	</p>
	<p align="left">
	  For $g \in \mathcal G$, &ensp;$f(g)$ is called the <i>fitness</i> of genotype $g$.
	</p>
</section>

<section>
	<h3>The epistasis model</h3>
	$
	  f(x_1,\, \ldots,\, x_p) = \beta_0 + \sum_{i} x_i \beta_i
	  + \sum_{i \le j} x_i x_j \beta_{i,j}
	$
	<img data-src="images/fitness_shapes_2_loci_BPS_axis.png">
	
	Epistasis is the deviation from the additive expectation of allelic effects:
	$$e = f(00) + f(11) - \Big(f(01) + f(10)\Big)$$

	<aside class="notes">
	  Mention shapes, higher-order interaction detection, TCS, etc.
	</aside>
</section>

<section>
	<img data-src="/assets/fitlands_outline.png" width=60%>

	<aside class="notes">
	  This is SotA pre AI in Bio revolution -- AF(1) came out in 2018.
	</aside>
</section>

<section>
	<h4>State of the art in 2018</h4>
	<img data-src="/assets/GlinternetVSxyz.svg" width=80%></br>
	<font size="3">
	  <a href="https://doi.org/10.1371/journal.pone.0254491">
		Kieran Elmes et al. Learning epistatic gene interactions from perturbation screens. 2021
	  </a>
	</font>

	<aside class="notes">
	  AF was published in 2018 (1), 2020 (2), 2024 (3).<br>
	  Conclusion: Glinternet is in, xyz is out.
	</aside>
</section>

<section>
	<h4>The hierarchy assumptions</h4>
	<img data-src="https://upload.wikimedia.org/wikipedia/commons/4/4f/Manhattan_plot_from_a_GWAS_of_kidney_stone_disease.png"><br>
	<font size="3">
	  <a href="https://doi.org/10.1038/s41467-019-13145-x">
	  Sarah Howles et al. Genetic variants of calcium and vitamin D metabolism in kidney stone disease. 2019</a>
	</font>
</section>

<section>
	<h4>Scalability</h4>
	<img data-src="/assets/glinternet_xyz_runtime.svg"><br>
	<font size="3">
	  <a href="https://doi.org/10.1371/journal.pone.0254491">
		Kieran Elmes et al. Learning epistatic gene interactions from perturbation screens. 2021
	  </a>
	</font>
</section>

<section>
	<font size="6">
	  Hierarchy and scalability are largely solved (for now)
	</font>
	<img data-src="/assets/pint_summary.svg"><br>
	<font size="3">
	  <a href="https://doi.org/10.1371/journal.pcbi.1010730">
		Kieran Elmes et al. A fast lasso-based method for inferring higher-order interactions. 2022
	  </a>
	</font>
</section>

<section>
  <section>
	<h3>Model assumptions</h3>
	$
	  f(x_1,\, \ldots,\, x_p) = \beta_0 + \sum_{i} x_i \beta_i
	  + \sum_{i \le j} x_i x_j \beta_{i,j}
	$
	$
	  + \sum_{i \le j \le k} x_i x_j x_k \beta_{i,j,k}
	  + \ldots
	$<br>
	<font size="3">
	  <a href="https://doi.org/10.1371/journal.pcbi.1010730">
		Kieran Elmes et al. A fast lasso-based method for inferring higher-order interactions. 2022
	  </a>
	</font>
  </section>
	
  <section>
	<h4>Why is fitness hard to measure?</h4>
	<div align="center">
	<img data-src="images/disc_diffusion.png" align="center">
	</div>
	<div align="center">
	<font size="3"><a href="https://en.wikipedia.org/wiki/Agar_diffusion_test">Wikipedia</a></font>
	</div>
  </section>
</section>

<section>
	<h4>Why don't you just use AI?</h4>
	<img data-src="/assets/linformer_model_selection.png" width=60%><br>
	What AI?

	<aside class="notes">
	  Transformer 2017, Linformer 2020
	</aside>
</section>

<section>
	<h3>SNVformer</h3>
	<img data-src="/assets/snvformer_predictions.svg"><br>
	<font size="3">
	  <a href="https://icml-compbio.github.io/2022/papers/WCBICML2022_paper_58.pdf">
		Kieran Elmes et al. SNVformer: an attention-based deep neural network for GWAS data. 2022
	  </a>
	</font>
</section>

<section>
	<img data-src="/assets/gene_graph_snvformer_encoder.svg" width="110%">
	<aside class="notes">
	  Optionally, genes can be encoded accounting for interaction reported in the BioGRID database
	</aside>
</section>

<section>
	<h3>Validation through simulations</h3>
	<ul>
	  <li> Just simulating from the epistasis (or higher-order interaction) model doesn't work&mdash;a liner approximation is typically good enough
	  <li> Can we simulate from a model for which a linear approximation fails?
	  <li> Can this model be also relevant?
	</ul>
</section>

<section>
	<h4>"eQTL" simulations summary</h4>
	$$
	  g \iff \beta_0 + \beta_1 v_1 + \dots + \beta_k v_k > 0
	$$
	<img data-src="/assets/eqtl_simulation_diagram.svg">
</section>

<section>
	<h4>Kolmogorov-Arnold Networks</h4>
	<img data-src="/assets/kan_theorem.jpg">
</section>

<section>
	<h4>Kolmogorov-Arnold Networks</h4>
	<img data-src="/assets/kan_diagram.jpg">
</section>

<section>
	<h4>KAN is better than the transformer? (unpublished work)</h4>
	<img data-src="/assets/KAN_for_GWAS_no_noise_summary_Lilin.svg">
	<p style="font-size:10pt" align="center">
	Lilin Zhang, Kieran Collienne, and Alex Gavryushkin (2024).
	</p>
</section>

<section>
	<h4>
	Notation
	</h4>
	For a genotype $g$, $w_g$ denotes the fitness of $g$,<br>
	for example, $w_{11}$ is the fitness of the double mutant.
	<h3>
	<p class="fragment" data-fragment-index="1">
	Epistasis &nbsp; $u_{11}$
	</p>
	</h3>
	<h4>
	<p class="fragment" data-fragment-index="1">
	Additive allelic effect &nbsp; $=$ &nbsp; no epistasis:
	</p>
	</h4>
	<p class="fragment" data-fragment-index="1">
	$$
	w_{11} + w_{00} = w_{01} + w_{10}
	$$
	<br>
	</p>
	<h4>
	<p class="fragment" data-fragment-index="2">
	Deviation from the additive expectation of allelic effects:
	<p>
	</h4>
	<p class="fragment" data-fragment-index="2">
	$u_{11} = w_{00} + w_{11} - (w_{01} + w_{10})$
	</p>
</section>

<section>
	<h3>Understanding three-way interactions</h3>

	<h4>
	<p class="fragment" data-fragment-index="1">
	Marginal epistasis?
	</p>
	</h4>
	<p class="fragment" data-fragment-index="1">
	$\small u_{\color{blue}{0}11} = w_{\color{blue}{0}00} + w_{\color{blue}{1}00} + w_{\color{blue}{0}11} + w_{\color{blue}{1}11} − (w_{\color{blue}{0}01} + w_{\color{blue}{1}01}) − (w_{\color{blue}{0}10} + w_{\color{blue}{1}10})$
	</p>

	<h4>
	<p class="fragment" data-fragment-index="2">
	Total three-way interaction?
	</p>
	</h4>
	 <p class="fragment" data-fragment-index="2">
	$\small u_{111} = w_{000} + w_{011} + w_{101} + w_{110} - (w_{001} + w_{010} + w_{100} + w_{111})$
	</p>

	<h4>
	<p class="fragment" data-fragment-index="3">
	Conditional epistasis?
	<p>
	</h4>
	<p class="fragment" data-fragment-index="3">
	$\small e = w_{\color{blue}{0}00} − w_{\color{blue}{0}01} − w_{\color{blue}{0}10} + w_{\color{blue}{0}11}$
	</p>
</section>

<section>
	<h3>Algebraic Geometry to the rescue</h3>
	$e = u_{011} + u_{111}$
	<p class="fragment" data-fragment-index="1">
	  In general, the four <i>interaction coordinates</i>
	  $$
	    u_{011}, u_{101}, u_{110}, u_{111}
	  $$
	  allow to describe <i>all</i> possible kinds of interaction!
	</p>
	<p class="fragment" data-fragment-index="2">
	  There are 20 types of interaction called <i>circuits</i> in Algebraic Geometry
	</p>
</section>

<section data-transition="slide-in none-out">
	$$
	\scriptsize
	\begin{align*}
	a&= w_{000}-w_{010}-w_{100}+w_{110} & m&=w_{001}+w_{010}+w_{100}-w_{111}-2w_{000}\\
	b&=w_{001}-w_{011}-w_{101}+w_{111} & n&=w_{011}+w_{101}+w_{110}-w_{000}-2w_{111}\\
	c&=w_{000}-w_{001}-w_{100}+w_{101} & o&=w_{010}+w_{100}+w_{111}-w_{001}-2w_{110}\\
	d&=w_{010}-w_{011}-w_{110}+w_{111} & p&=w_{000}+w_{011}+w_{101}-w_{110}-2w_{001}\\
	e&=w_{000}-w_{001}-w_{010}+w_{011} & q&=w_{001}+w_{100}+ w_{111}-w_{010}-2w_{101}\\
	f&=w_{100}-w_{101}-w_{110}+w_{111} & r&=w_{000}+w_{011}+ w_{110}-w_{101}-2w_{010}\\
	g&=w_{000}-w_{011}-w_{100}+w_{111} & s&=w_{000}+w_{101}+ w_{110}-w_{011}-2w_{100}\\
	h&=w_{001}-w_{010}-w_{101}+w_{110} & t&=w_{001}+w_{010}+w_{111}-w_{100}-2w_{011}\\
	i&=w_{000}-w_{010}-w_{101}+w_{111}\\
	j&=w_{001}-w_{011}-w_{100}+w_{110}\\
	k&=w_{000}-w_{001}-w_{110}+w_{111}\\
	l&=w_{010}-w_{011}-w_{100}+w_{101}\\
	\end{align*}
	$$
</section>

<section>
  <section>
	<h3>Competition experiments</h3>
	<p align="center">
	For example, malaria drug resistance:
	<img class="stretch" data-src="images/tem.svg" align="center">
	<br>
	<font size="3">Ogbunugafor et al. <i>Malar. J.</i> 2016</font>
	</p>
  </section>

  <section>
	<h4>Why is fitness hard to measure?</h4>
	<div align="center">
	<img data-src="images/disc_diffusion.png" align="center">
	</div>
	<div align="center">
	<font size="3"><a href="https://en.wikipedia.org/wiki/Agar_diffusion_test">Wikipedia</a></font>
	</div>
  </section>
</section>

<section>
	<h3>When partial order implies interaction?</h3>
	<br>
	We say that a partial order $\prec$ on the set of genotypes $\mathcal G = \{g_1,\ldots, g_m\}$ implies positive $f$-interaction if
	\[
	f(w_{g_1},\ldots, w_{g_m}) > 0
	\]
	whenever the genotypes $\mathcal G$ are ordered according to $\prec$.
</section>

<section>
	<h4>Precision oncology</h4>
	<img data-src="/assets/precision_oncology_by_vicka.svg">
</section>

<section>
	<img class="stretch" data-src="images/multiregion_tumour_samples.png">
	<p style="font-size:10pt" align="center">
	Alves, Joao, Sonia Prado-Lopez, Jose Manuel Cameselle-Teijeiro, David Posada. "Rapid evolution and biogeographic spread in a colorectal cancer" <i>bioRxiv</i> (2019).
	</p>
</section>

<section>
	<h3>Timing tumour evolution</h3>
	<img class="stretch" data-src="images/Sottoriva_tumour_time_tree.png">
	<p style="font-size:10pt" align="center">
	Lote, H., I. Spiteri, L. Ermini, A. Vatsiou, A. Roy, A. McDonald, N. Maka, et al. 2017. "Carbon Dating Cancer: Defining the Chronology of Metastatic Progression in Colorectal Cancer." <i>Annals of Oncology</i> 28 (6): 1243–49.
	</p>
</section>

<section data-background-image="./images/what_is_cancer_evo_problem.svg">
</section>

<section>
	<h4>Phylogenetic trees on tumor cells</h4>
	<img data-src="./images/tumor_cell_tree.svg" width=80%>
</section>

<section data-background-image="./images/multiomic_cancer_evo.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/spatial_phylo_on_all_cells.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/pathologist_annotated_cells.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/twenty_quality_cells.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/cells_with_same_genotypes.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/tree_on_all_quality_cells_expression.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/tree_on_cells_with_same_genotype_expression.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/tree_on_cells_with_same_genotype_snv.svg"
	data-background-size=80%>
</section>

<section data-background-image="./images/tree_on_cells_with_same_genotype_expression_snv.svg"
	data-background-size=80%>
</section>

<section>
	<h3>The big challenge: scalable algorithms for reconstructing evolutionary histories from sequence data</h3>
</section>

<section>
	<h3>NNI graph</h3>
	<img class="stretch" data-src="images/NNI.pdf.png">
</section>

<section>
	<h3>Theorem (phylogenetic classic)</h3>
	Computing the NNI distance is NP-hard.
</section>

<section>
	<h3>Discrete time-tree space</h3>
	<img class="stretch" data-src="images/DtT_with_rank_move.pdf.png">
</section>

<section>
	<h3>Theorem (Collienne, G)</h3>
	There exists an efficient (polynomial) algorithm for computing the RNNI distance.<br>
	<img data-src="/assets/lena_hatherton.jpg" width="60%">
</section>

<section>
	<h4>Geodesic in time-tree space</h4>
	<div style="font-size:10pt">
	<ul>
	  <li> Gavryushkin, Alex, and Alexei J. Drummond. "The space of ultrametric phylogenetic trees."
	  <i>Journal of Theoretical Biology</i> 403 (2016): 197-208.
	  <li> Gavryushkin, Alex, and Alexei Drummond. <i>tauGeodesic</i>. Mar. 2015. doi: 10.5281/zenodo.47152.
	<a href="https://github.com/gavruskin/tauGeodesic">https://github.com/gavruskin/tauGeodesic</a>
	</ul>
	<img data-src="images/trees.gif" width="60%">
	<div>
</section>

<section data-transition="fade-in slide-out" data-background-image="images/penguin_emperor_background.svg">
	<div style="font-size:10pt">
	<ul>
	  <li> Gavryushkina, Alexandra, Tracy A. Heath, Daniel T. Ksepka, Tanja Stadler,
	  David Welch, and Alexei J. Drummond.
	  "Bayesian total-evidence dating reveals the recent crown radiation of penguins."
	  <i>Systematic Biology</i> 66.1 (2016): 57-73.
	  <li> Gavryushkin, Alex, and Alexei Drummond. <i>tauGeodesic</i>. Mar. 2015. doi: 10.5281/zenodo.47152.
	<a href="https://github.com/gavruskin/tauGeodesic">https://github.com/gavruskin/tauGeodesic</a>
	</ul>
	<img data-src="images/trees_penguins.gif" width="60%">
	<div>
</section>

<section>
	<h4><a href="https://biods.org">Biological Data Science Lab</a></h4>
	<img data-src="/assets/2023-02-17-bioDS_lab.jpg" width=75%>
</section>

<section>
	<h3>Thanks for listening!</h3>
	<img data-src="/assets/2025-02-20-nsw_logo_proudly_funded.jpg" width=60%>
	<p>
	  <a href="https://linkedin.com/in/alex-gavryushkin">https://linkedin.com/in/alex-gavryushkin</a><br>
	  <a href="malto:alex@rnafold.ai">alex@rnafold.ai</a>
	</p>
</section>

</div>
</div>


<script src="../talks/revealjs/dist/reveal.js"></script>
<script src="../talks/revealjs/plugin/markdown/markdown.js"></script>
<script src="../talks/revealjs/plugin/highlight/highlight.js"></script>
<script src="../talks/revealjs/plugin/notes/notes.js"></script>
<script src="../talks/revealjs/plugin/math/math.js"></script>
<script>
	Reveal.initialize({
		controls: true,
		progress: true,
		history: true,
		center: true,
		slideNumber: 'h.v',
		plugins: [ RevealMarkdown, RevealHighlight, RevealNotes, RevealMath.MathJax2 ]
			});
</script>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-86783531-1', 'auto');
  ga('send', 'pageview');
</script>
</body>
</html>
